src/Repository/PersonRepository.php

```diff
@@ -89,4 +90,18 @@ public function sendRegisteredEmail(Person $person): bool
     {
         return $this->mailerClient->sendEmail($person->toMailerPayload());
     }
+
+    public function upgradePasswordIfPossible(?string $raw_password, Person $person): void
```

Comment:
IF we're going to throw let's put the precondition in a docblock:

```suggestion
    /**
     * Generates and persists a new password hash for this person if our existing hash for them wasn't made
     * using our current algorithm and settings.
     *
     * Must only be called for a person who has a password.
     */
    public function upgradePasswordIfPossible(?string $raw_password, Person $person): void
```